docs/csharp/language-reference/operators/bitwise-and-shift-operators.md

```diff
@@ -0,0 +1,182 @@
+
+The high-order empty bit positions are set based on the type of the first operand as follows:
+
+- If the first operand is of type [int](../keywords/int.md) or [long](../keywords/long.md), the right-shift operator performs an *arithmetic* shift: the value of the most significant bit (the sign bit) of the first operand is propagated to the high-order empty bit positions. That is, the high-order empty bit positions are set to zero if the first operand is non-negative and set to one if it's negative.
+
+  [!code-csharp-interactive[arithmetic right shift](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#ArithmeticRightShift)]
+
+- If the first operand is of type [uint](../keywords/uint.md) or [ulong](../keywords/ulong.md), the right-shift operator performs a *logical* shift: the high-order empty bit positions are always set to zero.
+
+  [!code-csharp-interactive[logical right shift](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#LogicalRightShift)]
+
+For information about how the second operand of the `>>` operator defines the shift count, see the [Shift count of the shift operators](#shift-count-of-the-shift-operators) section.
+
+## Logical AND operator &amp;
+
+The `&` operator computes the bitwise logical AND of its operands:
+
+[!code-csharp-interactive[bitwise AND](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#BitwiseAnd)]
+
+For the operands of the `bool` type, the `&` operator computes the [logical AND](boolean-logical-operators.md#logical-and-operator-) of its operands. The unary `&` operator is the [address-of operator](and-operator.md#unary-address-of-operator).
+
+## Logical exclusive OR operator ^
+
+The `^` operator computes the bitwise logical exclusive OR, also known as the bitwise logical XOR, of its operands:
+
+[!code-csharp-interactive[bitwise XOR](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#BitwiseXor)]
+
+For the operands of the `bool` type, the `^` operator computes the [logical exclusive OR](boolean-logical-operators.md#logical-exclusive-or-operator-) of its operands.
+
+## Logical OR operator |
+
+The `|` operator computes the bitwise logical OR of its operands:
+
+[!code-csharp-interactive[bitwise OR](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#BitwiseOr)]
+
+For the operands of the `bool` type, the `|` operator computes the [logical OR](boolean-logical-operators.md#logical-or-operator-) of its operands.
+
+## Compound assignment
+
+For a binary operator `op`, a compound assignment expression of the form
+
+```csharp
+x op= y
+```
+
+is equivalent to
+
+```csharp
+x = x op y
+```
+
+except that `x` is only evaluated once.
+
+The following example demonstrates the usage of compound assignment with the bitwise and shift operators:
```

Comment:
Is the article necessary here: "..with **the** bitwise and shift operators:"?
I've used it as the text is about defined five operators. If current version is correct, then I'll need to update other overviews with "the arithmetic operators", "the logical operators".

Reply:
I'd prefer removing it.